File: src/main/java/com/procurement/access/controller/LotsController.java
package com.procurement.access.controller;

import com.procurement.access.model.dto.bpe.ResponseDto;
import com.procurement.access.model.dto.lots.LotsRequestDto;
import com.procurement.access.model.dto.ocds.TenderStatus;
import com.procurement.access.model.dto.ocds.TenderStatusDetails;
import com.procurement.access.service.LotsService;
import javax.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

@Validated
@RestController
@RequestMapping("/lots")
public class LotsController {

    private final LotsService lotsService;

    public LotsController(final LotsService lotsService) {
        this.lotsService = lotsService;
    }

    @GetMapping
    public ResponseEntity<ResponseDto> getLots(@RequestParam("identifier") final String cpId,
                                               @RequestParam("stage") final String stage,
                                               @RequestParam("status") final String status) {
        return new ResponseEntity<>(
                lotsService.getLots(cpId, stage, TenderStatus.fromValue(status)),
                HttpStatus.OK);
    }

    @PostMapping("/updateStatus")
    public ResponseEntity<ResponseDto> updateStatus(@RequestParam("identifier") final String cpId,
                                                    @RequestParam("stage") final String stage,
                                                    @RequestParam("status") final String status,
                                                    @Valid @RequestBody final LotsRequestDto data) {
        return new ResponseEntity<>(
                lotsService.updateStatus(cpId, stage, TenderStatus.fromValue(status), data),
                HttpStatus.OK);
    }

    @PostMapping("/updateStatusDetails")
    public ResponseEntity<ResponseDto> updateStatusDetails(@RequestParam("identifier") final String cpId,
                                                           @RequestParam("stage") final String stage,
                                                           @RequestParam("statusDetails") final String statusDetails,
                                                           @Valid @RequestBody final LotsRequestDto data) {
        return new ResponseEntity<>(
                lotsService.updateStatusDetails(cpId, stage, TenderStatusDetails.fromValue(statusDetails), data),
                HttpStatus.OK);
    }
}

File: src/main/kotlin/com/procurement/access/controller/LotsController.kt
package com.procurement.access.controller

import com.procurement.access.service.LotsService
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@Validated
@RestController
@RequestMapping("/lots")
class LotsController(private val lotsService: LotsService) {

//    @GetMapping
//    fun getLots(@RequestParam("cpid") cpId: String,
//                @RequestParam("stage") stage: String,
//                @RequestParam("status") status: String): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.getLots(
//                        cpId = cpId,
//                        stage = stage,
//                        status = TenderStatus.fromValue(status)),
//                HttpStatus.OK)
//    }

//    @PostMapping("/updateStatusDetails")
//    fun updateStatusDetails(@RequestParam("cpid") cpId: String,
//                            @RequestParam("stage") stage: String,
//                            @RequestParam("statusDetails") statusDetails: String,
//                            @Valid @RequestBody data: UpdateLotsRq): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.updateStatusDetails(
//                        cpId = cpId,
//                        stage = stage,
//                        tenderStatusDetails = TenderStatusDetails.fromValue(statusDetails),
//                        lotsDto = data),
//                HttpStatus.OK)
//    }

//    @PostMapping("/updateStatusDetailsById")
//    fun updateStatusDetailsById(@RequestParam("cpid") cpId: String,
//                                @RequestParam("stage") stage: String,
//                                @RequestParam("lotAwarded") lotAwarded: Boolean,
//                                @RequestParam("lotId") lotId: String): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.updateStatusDetailsById(
//                        cpId = cpId,
//                        stage = stage,
//                        lotId = lotId,
//                        lotAwarded = lotAwarded),
//                HttpStatus.OK)
//    }

//    @PostMapping("/checkStatusDetailsGetItems")
//    fun checkStatusDetailsGetItems(@RequestParam("cpid") cpId: String,
//                           @RequestParam("stage") stage: String): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.checkStatusDetailsGetItems(
//                        cpId = cpId,
//                        stage = stage),
//                HttpStatus.OK)
//    }

//    @PostMapping("/checkStatus")
//    fun checkStatus(@RequestParam("cpid") cpId: String,
//                    @RequestParam("stage") stage: String,
//                    @Valid @RequestBody data: CheckLotStatusRq): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.checkStatus(
//                        cpId = cpId,
//                        stage = stage,
//                        lotDto = data),
//                HttpStatus.OK)
//    }

//    @PostMapping("/updateLots")
//    fun updateLots(@RequestParam("cpid") cpId: String,
//                   @RequestParam("stage") stage: String,
//                   @Valid @RequestBody data: UpdateLotsRq): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.updateLots(
//                        cpId = cpId,
//                        stage = stage,
//                        lotsDto = data),
//                HttpStatus.OK)
//    }

//    @PostMapping("/updateLotsEv")
//    fun updateLotsEv(@RequestParam("cpid") cpId: String,
//                     @RequestParam("stage") stage: String,
//                     @Valid @RequestBody data: UpdateLotsRq): ResponseEntity<ResponseDto> {
//        return ResponseEntity(
//                lotsService.updateLotsEv(
//                        cpId = cpId,
//                        stage = stage,
//                        lotsDto = data),
//                HttpStatus.OK)
//    }

}
